### ISSUE TITLE
Resource patterns with wildcard '*' results in failing test for parse_path

### ISSUE THREAD

**Issue by busunkim96** [googleapis/gapic-generator-python#701]
Asset has a resource path annotated with pattern `*`

https://github.com/googleapis/googleapis/blob/f1e0ce4dbc5f59448f441093014d712af442984f/google/cloud/asset/v1/assets.proto#L96-L100

```proto
message Asset {
  option (google.api.resource) = {
    type: "cloudasset.googleapis.com/Asset"
    pattern: "*"
  };
```

This results in a unhelpful set of resource name helpers: 
```py
    @staticmethod
    def asset_path() -> str:
        """Return a fully-qualified asset string."""
        return "*".format()

    @staticmethod
    def parse_asset_path(path: str) -> Dict[str, str]:
        """Parse a asset path into its component segments."""
        m = re.match(r"^*$", path)
        return m.groupdict() if m else {}

```

The generated tests break because of the `*`.
```py
def test_asset_path():
    expected = "*".format()
    actual = AssetServiceClient.asset_path()
    assert expected == actual


def test_parse_asset_path():
    expected = {}
    path = AssetServiceClient.asset_path(**expected)

    # Check that the path construction is reversible.
    actual = AssetServiceClient.parse_asset_path(path)
    assert expected == actual

```

```
                if not item or item[0][0] is AT:
>                   raise source.error("nothing to repeat",
                                       source.tell() - here + len(this))
E                   re.error: nothing to repeat at position 1

../../.pyenv/versions/3.8.1/lib/python3.8/sre_parse.py:668: error
```

The `*` pattern is uncommon and this method isn't actually helpful for parsing paths. I'm planning on deleting it from the generated output for now.

**Comment by busunkim96** [comment]
Similar issue with monitoring which has a path with `**`. 

https://github.com/googleapis/googleapis/blob/69697504d9eba1d064820c3085b4750767be6d08/google/monitoring/v3/metric_service.proto#L39-L46

```proto
option (google.api.resource_definition) = {
  type: "monitoring.googleapis.com/MetricDescriptor"
  pattern: "projects/{project}/metricDescriptors/{metric_descriptor=**}"
  pattern: "organizations/{organization}/metricDescriptors/{metric_descriptor=**}"
  pattern: "folders/{folder}/metricDescriptors/{metric_descriptor=**}"
  pattern: "*"
  history: ORIGINALLY_SINGLE_PATTERN
};
```

```py
    def metric_descriptor_path(project: str,) -> str:
        """Return a fully-qualified metric_descriptor string."""
        return "projects/{project}/metricDescriptors/{metric_descriptor=**}".format(
            project=project,
        )

    @staticmethod
    def parse_metric_descriptor_path(path: str) -> Dict[str, str]:
        """Parse a metric_descriptor path into its component segments."""
        m = re.match(
            r"^projects/(?P<project>.+?)/metricDescriptors/{metric_descriptor=**}$",
            path,
        )
        return m.groupdict() if m else {}
```

```py
def test_metric_descriptor_path():
    project = "squid"

    expected = "projects/{project}/metricDescriptors/{metric_descriptor=**}".format(
        project=project,
    )
    actual = MetricServiceClient.metric_descriptor_path(project)
    assert expected == actual


def test_parse_metric_descriptor_path():
    expected = {
        "project": "clam",
    }
    path = MetricServiceClient.metric_descriptor_path(**expected)

    # Check that the path construction is reversible.
    actual = MetricServiceClient.parse_metric_descriptor_path(path)
    assert expected == actual
```

**Comment by vam-google** [comment]
@busunkim96 Is this issue still relevant?

**Comment by busunkim96** [comment]
This was fixed by #1089.
